recipe/build-cpp.bat

```diff
@@ -15,6 +10,7 @@ cd build-cpp
 
 cmake ..  ^
```

Comment:
very likely. I feel like `-std=c14` is the wrong syntax and we may have to patch that up. But lets test.

Reply:
My understanding is that we need c++14 not 11. But i added a similar line.

Reply:
> My understanding is that we need c++14 not 11. But i added a similar line.

No, you need c++17, see [here](https://github.com/conda-forge/abseil-cpp-feedstock/blob/main/recipe/build.sh#L11) (as well as the evergreen: https://github.com/conda-forge/abseil-cpp-feedstock/issues/29)

Reply:
Ah sorry, this is windows. We're not setting the standard in https://github.com/conda-forge/abseil-cpp-feedstock/blob/main/recipe/bld.bat, which means it picks up whatever's the default upstream (the issue reference is still relevant though).

Reply:
> do you know what `LNK2001` is?

That's a signature mismatch cause by compiling against the wrong standard (because the classes don't match up, since C++11 doesn't have `std::basic_string` and needs to emulate it).

Reply:
> which means it picks up whatever's the default upstream.

20220623 still [uses](https://github.com/abseil/abseil-cpp/blob/20220623.0/CMake/AbseilHelpers.cmake#L264) C++11 by default (and is the last abseil release to do so)

Reply:
I think you may be correct on this hunch, but i think it is also related to the python compialtion part

Reply:
h-vetinari, can you push to my branch? it might be the easiest thing since you seem to know what to do.

Reply:
I'm very sure. I spent quite a while fighting with these very issues on the abseil feedstock, you can see the sad story in the dev branch.

Reply:
But the fix is trivial - just specify `-DCMAKE_CXX_STANDARD=11`

Reply:
understood. I might be able to revert the compiler change in this case too. Even if I do agree that moving to vs2019 is likely the good move....

Reply:
This doesn't seem to be true anymore:

```
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:234 (message):
  CMAKE_CXX_STANDARD is less than 14, please specify at least
  SET(CMAKE_CXX_STANDARD 14)
```

Reply:
> This doesn't seem to be true anymore:

That error is coming from grpc, not abseil. And you need to compile everything against the same cxx standard, which upstream has no problem doing.